xarray/core/dataarray.py

```diff
@@ -2381,15 +2404,20 @@ def to_dataframe(self, name: Hashable = None) -> pd.DataFrame:
                 "DataFrame: use the ``name`` parameter"
             )
 
-        dims = dict(zip(self.dims, self.shape))
         # By using a unique name, we can convert a DataArray into a DataFrame
         # even if it shares a name with one of its coordinates.
         # I would normally use unique_name = object() but that results in a
         # dataframe with columns in the wrong order, for reasons I have not
         # been able to debug (possibly a pandas bug?).
         unique_name = "__unique_name_identifier_z98xfz98xugfg73ho__"
         ds = self._to_dataset_whole(name=unique_name)
-        df = ds._to_dataframe(dims)
+
+        if dim_order is None:
```

Comment:
can we handle the `None` case in `_normalize_dim_order`?

Reply:
We need it here because `DataArray.to_dataframe()` uses a different dimension order than `Dataset.to_dataframe()`

Reply:
oh thanks!